### ISSUE TITLE
"Splat not found" Error when compiling test program for ported lib

### ISSUE THREAD

**Issue by thibmo** [beefytech/Beef#192]
When compiling the project I instantly recieve the following error dialog:
![Error dlg](https://i.imgur.com/rWksva2.png)

The lib itself only has a .dll or .so

If more info is needed, just shoot. :)
[lua535_test.zip](https://github.com/beefytech/Beef/files/4568360/lua535_test.zip)

**Comment by thibmo** [comment]
Btw, for an updated lib: https://github.com/thibmo/lua535-beef
And for the updated test: https://github.com/thibmo/lua535-beef-test

**Comment by bfiete** [comment]
Okay- so you did uncover a real compiler bug, which I'm fixing, but here's what was happening:
```
public static String tostring(lua_State* L, int32 i)
{
	var result = "";
        tolstring(L, i, null).ToString(result);
	return result;
}
```

`tolstring` is returning a char8*. You'r calling ToString on that, which is actually causing the char8* to be boxed as a System.Pointer<char8> and then the ToString method is called on that which would just output the address.

You are attempting to write the string into the string literal `""`, which is read-only memory and will crash. Someone needs to allocate a string.

So you could do this
```
public static String tostring(lua_State* L, int32 i)
{
    return new String(tolstring(L, i, null));
}
```

but you probably really want the tostring method to take the desired string buffer as an argument
```
public static String tostring(lua_State* L, int32 i, String outString)
{
    outString.Append(tolstring(L, i, null));
}
```

**Comment by thibmo** [comment]
Thanks for explaining, I'll get on fixing it :)

**Comment by bfiete** [comment]
This has been fixed, will be in 0.42.4.
